### PR TITLE
[feature][connector-v2][clickhouse] Support write cdc changelog event in clickhouse sink

### LINKED ISSUE #3652: [Feature][CDC] Support RowKind(DELETE/UPDATE_BEFORE/UPDATE_AFTER) data row in Clickhouse sink
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Support RowKind(DELETE/UPDATE_BEFORE/UPDATE_AFTER) data row in Clickhouse sink

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hailin0**
## Purpose of this pull request

Close #3652

MergeTree Table Engine:
- Support enable `allowExperimentalLightweightDelete` setting
- CDC Events:
  - INSERT: INSERT INTO SQL
  - UPDATE_BEFORE: DELETE SQL
  - UPDATE_AFTER: INSERT INTO SQL
  - DELETE: DELETE SQL

- ReplacingMergeTree Engine CDC Events:
  - INSERT: INSERT INTO SQL
  - UPDATE_BEFORE: ignore
  - UPDATE_AFTER: INSERT INTO SQL
  - DELETE: DELETE SQL

Other Table Engine:
- CDC Events:
  - INSERT: INSERT INTO SQL
  - UPDATE_BEFORE: ignore
  - UPDATE_AFTER: ALTER TABLE UPDATE SQL
  - DELETE: ALTER TABLE DELETE SQL

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Comment by hailin0** [comment]
@EricJoy2048 @Hisoka-X PTAL

File: seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/sink/client/executor/BufferReducedBatchStatementExecutor.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+
+@AllArgsConstructor
+@RequiredArgsConstructor
+public class BufferReducedBatchStatementExecutor implements JdbcBatchStatementExecutor {
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/sink/client/executor/BufferReducedBatchStatementExecutor.java]: Maybe change name for `BufferedBatchStatementExecutor` and `BufferReducedBatchStatementExecutor`? Hard to know difference between two executor.

File: seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/sink/client/executor/InsertOrUpdateBatchStatementExecutor.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        boolean currentChangeFlag = hasInsert(record);
+        if (currentChangeFlag) {
+            if (preChangeFlag != null && !preChangeFlag) {
+                insertStatement.executeBatch();
+                insertStatement.clearBatch();
+            }
+            valueRowConverter.toExternal(record, updateStatement);
+            updateStatement.addBatch();
+        } else {
+            if (preChangeFlag != null && preChangeFlag) {
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/sink/client/executor/InsertOrUpdateBatchStatementExecutor.java]: If the previous record is an update and this record is an insert, it seems unnecessary to execute the update statement.

**Review by EricJoy2048 — APPROVED** [review]

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Pull request merged**